### PR TITLE
Adds IPC themed helmets.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

This PR adds IPC themed television helmets to the game. Television helmets can be crafted by taking a status display screen, unhooked from a wall, and hollowing it out with a screwdriver and crowbar in the crafting menu. One can also be purchased as contraband through the costume vendor. They have snug fit to prevent them from being knocked off through hat-tossing, and due to the nature of light reflecting through a television screen, wearing one make the user more sensitive to being flashed. 

image

Additionally, due to be a whole TV that you hollowed out and stuck on your head, it comes with an FOV cone.

## Why It's Good For The Game

By headcoder decision, IPCs have been blocked from being added to /tg/ due to the overlap between what IPCs would add, and what borgs already do. For all intents and purposes, that's a settled issue for the moment. However, as a middle ground, I wanted to add something flavorful as middle ground while still providing an interesting gameplay gimmick.

## Changelog

:cl:
add: Television themed headgear has arrived! Hack a costume vendor or hollow out an unsecured status display to make one. Just be careful not to get flashed while wearing one.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
I saw "Adds IPCs." and my first thought was GOOD LUCK.

This, however. This is good.

**Comment by san7890** [comment]
Admin spawnable subtype without the FOV cone would be nice to have for events.

**Comment by TheVekter** [comment]
Are you still able to display things on them like with the IPC heads?

**Comment by ArcaneMusic** [comment]
> Are you still able to display things on them like with the IPC heads?

Not currently, as the TV is unplugged visually.

**Comment by Mooshimi** [comment]
may want to edit the status display sprite when it's not on the wall since "hollowing" it out would look like it's being inflated by the screwdriver as it currently is

**Comment by optimumtact** [comment]
>They have snug fit to prevent them from being knocked off through hat-tossing

why

**Comment by Fikou** [comment]
> > They have snug fit to prevent them from being knocked off through hat-tossing
> 
> why

its consistent with other headwear that fully covers your head, like cueball helmets, snowmen head, sec helmets etc

**Comment by optimumtact** [comment]
ok

**Comment by Shadowflame909** [comment]
Do you guys think wearing a hollowed out box tv on your neck would be heavy as heck. Maybe the frame was made out of plastic?

**Review by Imaginos16 — APPROVED** [review]
yea sure

File: code/datums/components/crafting/tailoring.dm [hunk 1]
```diff
@@ -164,3 +164,12 @@
 	reqs = list(/obj/item/stack/sheet/cloth = 2, /obj/item/stack/license_plates = 1)
 	time = 10
 	category = CAT_CLOTHING
+
+/datum/crafting_recipe/tv_helmet
+	name = "Television Helmet"
+	result = /obj/item/clothing/head/tv_head
+	tool_behaviors = list(TOOL_SCREWDRIVER, TOOL_CROWBAR)
+	reqs = list(/obj/item/wallframe/status_display = 1)
+	time = 20
```
Comment (Fikou) [inline code/datums/components/crafting/tailoring.dm:173]: use SECONDS defines

File: code/modules/clothing/head/costume.dm [hunk 2]
```diff
@@ -179,3 +179,25 @@
...
+	var/has_FOV = TRUE
+
+/obj/item/clothing/head/tv_head/Initialize(mapload)
+	. = ..()
+	if(has_FOV)
+		AddComponent(/datum/component/clothing_fov_visor, FOV_90_DEGREES)
+
+/obj/item/clothing/head/tv_head/fov_less
+	desc = "A mysterious headgear made from the hollowed out remains of a status display. How very retro-retro-futuristic of you. It's very easy to see out of this one."
+	has_FOV = FALSE
```
Comment (Mothblocks) [inline code/modules/clothing/head/costume.dm:203]:
```suggestion
	var/has_fov = TRUE

/obj/item/clothing/head/tv_head/Initialize(mapload)
	. = ..()
	if(has_fov)
		AddComponent(/datum/component/clothing_fov_visor, FOV_90_DEGREES)

/obj/item/clothing/head/tv_head/fov_less
	desc = "A mysterious headgear made from the hollowed out remains of a status display. How very retro-retro-futuristic of you. It's very easy to see out of this one."
	has_fov = FALSE
```

File: code/modules/clothing/head/costume.dm [hunk 3]
```diff
@@ -179,3 +179,25 @@
 	name = "basil boys helmet"
 	icon_state = "football_helmet_red"
 	inhand_icon_state = "football_helmet_red"
+
+/obj/item/clothing/head/tv_head
+	name = "television helmet"
+	desc = "A mysterious headgear made from the hollowed out remains of a status display. How very retro-retro-futuristic of you."
+	icon_state = "IPC_helmet"
+	inhand_icon_state = "syringe_kit"
+	lefthand_file = 'icons/mob/inhands/equipment/medical_lefthand.dmi' //Grandfathered in from the wallframe for status displays.
```
Comment (Mothblocks) [inline code/modules/clothing/head/costume.dm:188]: Dude what

Comment (Mothblocks) [reply code/modules/clothing/head/costume.dm:188]: Make a new file please

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/costume.dm (hunk 3 above)
Comment (Mothblocks) [reply code/modules/clothing/head/costume.dm:188]: Oh wait this is for hands

**Pull request merged**